js/runner.go

```diff
@@ -113,6 +116,97 @@ func newFromBundle(logger *logrus.Logger, b *Bundle) (*Runner, error) {
+                case '/':
+                        return ipPoolFromCidr(s)
+                }
+        }
+        return nil
+}
+
+func ipPoolFromRange(s string) *IpPool {
+        ss := strings.SplitN(s, "-", 2)
+        ip0, ip1 := net.ParseIP(ss[0]), net.ParseIP(ss[1]) // len(ParseIP())==16
+        if ip0 == nil || ip1 == nil {
+                fmt.Println("Wrong IP range format, exit.", s)
+                return nil
+        }
+        n0 := binary.BigEndian.Uint64(ip0[:8])
+        h0 := binary.BigEndian.Uint64(ip0[8:])
+        n1 := binary.BigEndian.Uint64(ip1[:8])
+        h1 := binary.BigEndian.Uint64(ip1[8:])
+        if (n0 > n1) || (h0 > h1) {
+                fmt.Println("Negative IP range, exit.", s)
+                return nil
+        }
+        return &IpPool{ip0, h0, n0, h1 - h0 + 1, n1 - n0 + 1}
+}
+
+func ipPoolFromCidr(s string) *IpPool {
+        ip0, pnet, err := net.ParseCIDR(s)
+        if err != nil {
+                fmt.Println("ParseCIDR() failed: ", s)
+                return nil
+        }
+        // ip0 (range start) >= ip1 (cidr start)
+        ip1 := pnet.IP.To16()
+        n0 := binary.BigEndian.Uint64(ip0[:8])
+        h0 := binary.BigEndian.Uint64(ip0[8:])
+        n1 := binary.BigEndian.Uint64(ip1[:8])
+        h1 := binary.BigEndian.Uint64(ip1[8:])
+        ones, bits := pnet.Mask.Size()
+        mask := bits - ones
+        hostPos, netPos := h0-h1, n0-n1
+        netRangeNum, hostRangeNum := uint64(0), uint64(1<<mask)-hostPos
+        if mask > 64 {
+                netRangeNum, hostRangeNum = uint64(1<<(mask-64))-netPos, ^uint64(0)-hostPos
+        }
+        return &IpPool{ip0, h0, n0, hostRangeNum, netRangeNum}
+}
+
+// GetRandomIP return a random-by-seed selected IP from an IP pool
+// IPv4 pool is considered as a uniform distribution from start IP to end IP
+// IPv6 poop is considered as two independent uniform distributions: high 64-bit and low 64-bit
+func (pool *IpPool) GetRandomIP(seed int64) net.IP {
+        var hostShift, netShift uint64
+        r := rand.New(rand.NewSource(seed))
+        if pool.hostRangeNum > 0 {
+                hostShift = r.Uint64() % pool.hostRangeNum
+        }
+        if b := pool.ip.To4(); b != nil {
+                i := pool.hostStart + hostShift
+                return net.IPv4(byte(i>>24), byte(i>>16), byte(i>>8), byte(i))
+        }
+        if pool.netRangeNum > 0 {
+                netShift = r.Uint64() % pool.netRangeNum
+        }
+        if b := pool.ip.To16(); b != nil {
+                netPos := pool.netStart + netShift
+                hostPos := pool.hostStart + hostShift
+                if hostPos < pool.hostStart {
+                        netPos = netPos + 1
+                }
+                if ip6 := make(net.IP, net.IPv6len); ip6 != nil {
+                        binary.BigEndian.PutUint64(ip6[:8], netPos)
+                        binary.BigEndian.PutUint64(ip6[8:], hostPos)
+                        return ip6
+                }
```

Comment:
I don't think I like that :(. 

You might have seen this in the wild and maybe this is "common" practice but given how ipv6 is supposed to be structured I find it very unlikely that you will get a mask that is ... THIS big :D 

And as I've argued before we probably should give IPs to VUs in some orderly way so that if you have 20 IPs and 40 VUs -  each IP will be used by 2 VUs which currently doesn't seem like something that will happen.

Reply:
this is used to test IPv6 only - both of network part and host part should has a range to change. typically, for load balance algorithm like RSS that requires a good hash key including both of network and host bytes. if you don't need this feature, just set a mask >= 64 or a small range.

Reply:
from my case, I often use 40K IPs with one Linux machine:)

Reply:
// IPv4: one IP block from start IP to end IP
// IPv6: two independent IP blocks: high 64-bit for net and low 64-bit for hosts
// This is useful if you want to travel some IPv6 nets without traveling 2^64 hosts

Reply:
As I explained above ... having 2^65 ips for *outgoing* requests ... seems impossible at least for anything that k6 is currently runnig on. Even at 40k IPs ... your OS is probably using more memory just to configure this that k6 will use to unroll them :). 

I might be missing some strange use case and way to do this but as far as I can see ... while you can having full subnet for incoming requests with [anyip](https://blog.widodh.nl/2016/04/anyip-bind-a-whole-subnet-to-your-linux-machine/) this seems to not work for outgoing ones ... which more or less removes the need for k6 to support subnets of the size of /65+, actually a lot less as I would expect the resource usage of a OS to overwhelm thsi pretty fast.

Also this is pretty impractical for k6 ... as you are unlikely to have enough VUs to use even a fraction of the IPs  on a single machine (I would really like it if it was possible to run 2^20 VUs .. but even with 1mb of memory per each this is like 1TB of memory ... just for the VUs which still means that the unrolling of the IPs will be a drop in the ocean of memory used

Reply:
local type route is also usable for k6 outgoing requests, so no need to setup IPs to linux kernel -- actually I was using it for a while using serval GB memory.

Reply:
here is my K6 using 4k VUs:
![image](https://user-images.githubusercontent.com/3319979/96014743-4ec07700-0e79-11eb-9318-b0b213c5050d.png)
![image](https://user-images.githubusercontent.com/3319979/96014775-57b14880-0e79-11eb-8d9d-1ac35a44495e.png)

Reply:
Can you elaborate on how you actually managed to get requests out of a /65 ipv6 network ... for example

Reply:
For example, a /65 IPv6 range format like fd00:1:2:0::0-fd00:1:2:ff::3ff means:
network will be range of 0 ~ 255 and selected by VU_id % IPBlock.netN
host will be range of 0 ~ 1023 and selected by VU_id % IPBlock.hostN

Of cause, if one don't want so strange range method(it is most like case to test load balancer), just give host part for range or give a CIDR mask <= /64

Reply:
I just now noticed that you have actually explained how to make the setup in the PR description :facepalm: 
I did manage to add /31 network (I only have ipv4 and didn't want to break my network configuring stuff to test this :D ) and it seems to work ... I am still somewhat skeptic that it will be a common practice to use that on load generator machine ... but it's fine to support it